src/output_modules/module_json.c

```diff
@@ -50,7 +50,7 @@ int json_output_file_init(struct state_conf *conf, UNUSED char **fields,
 	return EXIT_SUCCESS;
 }
 
-char *hex_encode(char *packet, int buflen)
+char *hex_encode(unsigned char *packet, int buflen)
```

Comment:
@zakird I come from the "bytes are unsigned 8-bit integers, not characters" camp.

@paul-pearce You're probably right about `%.2hhx`, but at the same time this is exactly why `printf` has been known to induce tears.